tortoise/backends/base/client.py

```diff
@@ -54,59 +58,146 @@ def __init__(
 
         super().__setattr__("_mutable", False)
 
-    def __setattr__(self, attr, value):
+    def __setattr__(self, attr: str, value: Any) -> None:
         if not getattr(self, "_mutable", False):
             raise AttributeError(attr)
-        return super().__setattr__(attr, value)
+        super().__setattr__(attr, value)
 
     def __str__(self) -> str:
         return str(self.__dict__)
 
 
 class BaseDBAsyncClient:
+    """
+    Base class for containing a DB connection.
+
+    Parameters get passed as kwargs, and is mostly driver specific.
+
+    .. attribute:: query_class
+        :annotation: Type[pypika.Query]
+
+        The PyPika Query dialect (low level dialect)
+
+    .. attribute:: executor_class
+        :annotation: Type[BaseExecutor]
+
+        The executor dialect class (high level dialect)
+
+    .. attribute:: schema_generator
+        :annotation: Type[BaseSchemaGenerator]
+
+        The DDL schema generator
+
+    .. attribute:: capabilities
+        :annotation: Capabilities
+
+        Contains the connection capabilities
+    """
+
```

Comment:
might be worth mentioning that none of this should be called directly but that the model methods should be used generally for people who end up here when searching for something like "insert" or "execute query"

Reply:
That would unfortunately require making querying SQL directly an official API. So this might be done better as an extra feature?

Reply:
i mean for general things, most people will wana do simple selects or inserts but if they just use the search to find "insert", this might come up